version: "3.8"

services:
  flask_app:
    build: ./app
    container_name: flask_app
    environment:
      - FLASK_ENV=production
    volumes:
      - ./app/static/uploads:/app/static/uploads
      - ./app/database:/app/database
    expose:
      - "5000"
    networks:
      - app_network
    command: ["python", "app.py"]
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:5000/"]
      interval: 30s
      retries: 3


  nginx:
    image: nginx:latest
    container_name: nginx_proxy
    volumes:
      - ./app/nginx.conf:/etc/nginx/conf.d/default.conf:ro
      - ./app/static/uploads:/app/static/uploads
      - ./logs/nginx:/var/log/nginx
    ports:
      - "80:80"
    depends_on:
      - flask_app
    networks:
      - app_network

networks:
  app_network:
    driver: bridge
